src/main/java/com/academy/edge/studentmanager/services/impl/StudentServiceImpl.java

```diff
@@ -93,6 +93,49 @@ public StudentResponseDTO insertStudent(StudentCreateDTO studentCreateDTO, Multi
         return modelMapper.map(student, StudentResponseDTO.class);
     }
 
+    @Override
+    public StudentResponseDTO updateStudent(String email, StudentUpdateDTO studentUpdateDTO) {
+        Student student = studentRepository
+                .findByEmail(email)
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Student not found with email: " + email));
+
+        modelMapper.map(studentUpdateDTO, student);
+        studentRepository.save(student);
+
+        return modelMapper.map(student, StudentResponseDTO.class);
+    }
+
+    @Override
+    public StudentResponseDTO updateStudentPhoto(String email, MultipartFile file) {
+        Student student = studentRepository
+                .findByEmail(email)
+                .orElseThrow(() -> new ResponseStatusException(NOT_FOUND, "Student not found with email: " + email));
+
+        if(!contentTypes.contains(file.getContentType())){
+            throw new ResponseStatusException(BAD_REQUEST, "File is not a image file");
+        }
+
+        String newPhotoUrl = student.getRegistration() + "_" + file.getOriginalFilename();
+        String oldPhotoUrl = student.getPhotoUrl();
+
+        if (newPhotoUrl.equals(student.getPhotoUrl())) {
+            newPhotoUrl = student.getRegistration() + "_new_" + file.getOriginalFilename();
+        }
+
+        try {
+            s3Service.uploadFile(newPhotoUrl, file);
+            s3Service.deleteFile(oldPhotoUrl);
+        } catch (IOException e) {
+            s3Service.deleteFile(newPhotoUrl);
+            throw new RuntimeException(e);
```

Comment:
Ao invés de um throw new RunTimeException, podemos mapear para um erro de servidor, o que acha? Como é feito na linha 112.

Reply:
Concordo. Mantive esse padrão por conta do que estava sendo executado em insertStudent, mas acredito que seja realmente melhor. Possivelmente ele já retornaria um erro 500, pois não tem nenhum handler para pegar essa exceção e o próprio spring iria pegar. Mas mapear para uma exceção mais específica é realmente o ideal. Tenho uma issue para o insertStudent, então quando eu for mexer nesse método já atualizo para mapear para essa exceção também.